org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java

```diff
@@ -52,7 +53,7 @@ public class CompletionProposalDescriptionProvider {
 	 */
 	private CompletionContext fContext;
 	private ICompilationUnit fUnit;
-	private Map<String, Integer> fCollapsedTypes;
+	private Map<String, Integer> fCollapsedTypes = new HashMap<>();
```

Comment:
you should probably instantiate an empty map (size 0) in the constructors, to avoid creating an unnecessary object in the most common case.